apis/nucleus/src/components/listbox/listbox-keyboard-navigation.js

```diff
@@ -0,0 +1,63 @@
+import KEYS from '../../keys';
+
+const getElement = (elm, next = false) =>
+  elm && elm.parentElement[next ? 'nextElementSibling' : 'previousElementSibling'].querySelector('[role]');
+
+export default function getKeyboardNavigation({ select, confirm, cancel }) {
+  let startedRange = false;
+  const setStartedRange = (val) => {
+    startedRange = val;
+  };
+
+  const handleKeyDown = (event) => {
+    let elementToFocus;
+    const { keyCode, shiftKey = false } = event.nativeEvent;
+
+    switch (keyCode) {
+      case KEYS.SHIFT:
+        // This is to ensure we include the first value when starting a range selection.
+        setStartedRange(true);
+        break;
+      case KEYS.SPACE:
+        select([+event.currentTarget.getAttribute('data-n')]);
+        break;
+      case KEYS.ARROW_DOWN:
+      case KEYS.ARROW_RIGHT:
+        elementToFocus = getElement(event.currentTarget, true);
+        if (shiftKey && elementToFocus) {
+          if (startedRange) {
+            select([+event.currentTarget.getAttribute('data-n')], true);
+            setStartedRange(false);
+          }
+          select([+elementToFocus.getAttribute('data-n')], true);
+        }
+        break;
+      case KEYS.ARROW_UP:
+      case KEYS.ARROW_LEFT:
+        elementToFocus = getElement(event.currentTarget, false);
+        if (shiftKey && elementToFocus) {
+          if (startedRange) {
+            select([+event.currentTarget.getAttribute('data-n')], true);
+            setStartedRange(false);
+          }
+          select([+elementToFocus.getAttribute('data-n')], true);
+        }
+        break;
+      case KEYS.ENTER:
+        confirm();
+        break;
+      case KEYS.ESCAPE:
+        cancel();
+        if (document.activeElement) {
+          document.activeElement.blur();
+        }
```

Comment:
In most real use cases I expect that you want to move the focus somewhere else (outside the listbox) and not just remove it. (but the listbox itself will not have enought information)

Reply:
Yeah, maybe this is where the `useKeyboard` can help. I think @Caele might have a look next week and see how we can use the keyboard. I am far from 100% sure about the accessibility best practices here so may be more holes to fill. We can improve iteratively.